### ISSUE TITLE
[OUDS] Color issue while using `.bg-*` helpers

### ISSUE THREAD

**Issue by louismaximepiton** [Orange-OpenSource/Orange-Boosted-Bootstrap#3012]
### Prerequisites

- [x] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Add a way to contextualize the text color usage while using `.bg-*`.

We thought about adding a CSS var, something like this:
```scss
.bg-* {
  --bs-local-color: $local-color;
  ...
}

[data-bs-theme] {
  color: var(--bs-local-color, var(--bs-color-content-default));
  ...
}
```

It looks a bit odd, but it works 🤷 

### Motivation and context

```html
<div class="text-bg-status-neutral-emphasized">
  <div class="container-fluid container-max-width py-scaled-xs" data-bs-theme="root-inverted">
    <div class="d-flex justify-content-center align-items-center gap-scaled-xl">
      <label for="switch1" class="h4 mb-none">Orange dla sektora publicz</label>
      <label class="switch-standalone p-none">
        <input id="switch1" class="control-item-indicator" type="checkbox" role="switch" value="" checked />
      </label>
    </div>
  </div>
</div>
```

This code has an issue since the `Orange dla sektora publicz` should be printed white but is right now printed light gray (`#eeeeee`).

```html
<div class="text-bg-status-neutral-emphasized">
  <div class="container-fluid container-max-width py-scaled-xs">
    <div class="d-flex justify-content-center align-items-center gap-scaled-xl">
      <label for="switch1" class="h4 mb-none">Orange dla sektora publicz</label>
      <label class="switch-standalone p-none" data-bs-theme="root-inverted">
        <input id="switch1" class="control-item-indicator" type="checkbox" role="switch" value="" checked />
      </label>
    </div>
  </div>
</div>
```

Resolves for now
